[skip ci] Update INSTALLING.md

bbeggsTT opened the pull request:
### Ticket
N/A

### Problem description
Installation procedure is outdated in many places. Too many links/clicks to get through the procedure.

### What's changed
Updated installation procedure for accuracy, to cut down on link hopping, and to be an independent module.

bbeggsTT committed:
Update INSTALLING.md

Updated installation procedure for accuracy, to cut down on link hopping, and to be an independent module.

bbeggsTT committed:
Merge branch 'main' into bbeggsTT-patch-5

dimitri-tenstorrent commented on INSTALLING.md:
````diff
@@ -2,199 +2,178 @@
+./build_metal.sh
 ```
 
-#### Option 3: From Docker Release Image
````
Is there a reason this section was deleted?

dimitri-tenstorrent replied:
I know Bryan and Jasmina were looking for this simplified option.

gsarabandoTT replied:
It shouldn't have been. We were seeking clarity for the multiple installing options, we'll reintegrate it asap.

dimitri-tenstorrent replied:
I can explain some of the advantages for this with Docker we do not need the following steps as they are already included.
`Step 2. System-level dependencies`
`Step 4. Install and start using TT-NN and TT-Metalium!`

dimitri-tenstorrent replied:
So it might require some re-work of the installation instructions since it does not match the mold.

blozano-tt commented on INSTALLING.md:
````diff
@@ -2,199 +2,178 @@
-| T3000 (Wormhole)    | Ubuntu 20.04    | 3.8.10   | v1.27.1            | fw_pack-80.10.0.0 (v80.10.0.0)             | v2.2.0 or above       | v1.1.3 or above, `mesh` config |
+#### Install System-level Dependencies
+```
+wget https://raw.githubusercontent.com/tenstorrent/tt-metal/refs/heads/main/install_dependencies.sh
````
What’s the reason for using wget instead of just cloning? (Not saying it’s wrong, just want to understand).

Also, might need to do the following now:

chmod a+x install_dependencies.s

to make the script executable by root.

gsarabandoTT replied:
this was an effort in simplifying the process (have fewer overall steps). The install_dependencies.sh also installs git and git-lfs which are necessary to cloning the repo. So the reasoning was providing an order of steps that provides every necessary component in order.

gsarabandoTT replied:
Thank you @blozano-tt, bash was taking care of that but chmod seems like a safer option. Included it in latest commit.

dimitri-tenstorrent commented on INSTALLING.md:
````diff
@@ -2,199 +2,178 @@
-When inside of the container,
+- For Blackhole:
 ```sh
-python3 -c "import ttnn"
````
Could we use the same verification step here instead of `python3 -c "import ttnn"` use `python3 -m ttnn.examples.usage.run_op_on_device`

gsarabandoTT replied:
`python3 -m ttnn.examples.usage.run_op_on_device` is suggested right after that. Is `python3 -c "import ttnn"` not a required step for the docker install?

dimitri-tenstorrent replied:
No it was a validation step as well, but I'd rather have the same validation step as for the wheel.

gsarabandoTT committed:
add wheel and docker installing options

gsarabandoTT committed:
update install_dependencies.sh permission

dimitri-tenstorrent commented:
No not required.

dimitri-tenstorrent commented on INSTALLING.md:
````diff
@@ -2,199 +2,262 @@
+  docker pull ghcr.io/tenstorrent/tt-metal/tt-metalium-ubuntu-20.04-amd64-release/wormhole_b0:latest-rc
+  docker run --it --rm -v /dev/hugepages-1G:/dev/hugepages-1G --device /dev/tenstorrent ghcr.io/tenstorrent/tt-metal/tt-metalium-ubuntu-20.04-amd64-release/wormhole_b0:latest-rc bash
+  ```
+  - For Blackhole:
````
Technically, we do not build an image for the blackhole. Could we remove this section?

gsarabandoTT replied:
The whole section or just the run command?

dimitri-tenstorrent replied:
We do not build docker images for blackhole. I can add a TODO to add that

dimitri-tenstorrent commented on INSTALLING.md:
````diff
@@ -2,199 +2,262 @@
+  - For Grayskull:
+  ```
+  docker pull ghcr.io/tenstorrent/tt-metal/tt-metalium-ubuntu-20.04-amd64-release/grayskull:latest-rc
+  docker run --it --rm -v /dev/hugepages-1G:/dev/hugepages-1G --device /dev/tenstorrent ghcr.io/tenstorrent/tt-metal/tt-metalium-ubuntu-20.04-amd64-release/grayskull:latest-rc bash
````
For all of the three cards, there is a mistake (by me) where I use `--it` instead of `-it`. One dash only.

gsarabandoTT committed:
remove commands not yet supported

dimitri-tenstorrent reviewed (approved):
Approved for the Docker Release sections :) Thank you!

dimitri-tenstorrent commented on INSTALLING.md:
```diff
@@ -2,199 +2,266 @@
-| Grayskull           | Ubuntu 20.04    | 3.8.10   | v1.27.1            | fw_pack-80.9.0.0 (v80.9.0.0)               | v2.2.0 or above       | N/A                            |
-| Wormhole            | Ubuntu 20.04    | 3.8.10   | v1.27.1            | fw_pack-80.10.0.0 (v80.10.0.0)             | v2.2.0 or above       | N/A                            |
-| T3000 (Wormhole)    | Ubuntu 20.04    | 3.8.10   | v1.27.1            | fw_pack-80.10.0.0 (v80.10.0.0)             | v2.2.0 or above       | v1.1.3 or above, `mesh` config |
+#### Install System-level Dependencies
```
Just to be clear these are not required for Docker Release install

gsarabandoTT replied:
I kept it there for now because many of these dependencies are required during the driver and firmware installation. I believe having it as a first step makes things flow easier. On the other hand we should definitely look for better alternatives to wrap things together. Let me know if keeping it there is not an option.

tt-rkim commented on INSTALLING.md:
```diff
@@ -2,199 +2,266 @@
 ---
 
-### Step 2. System-level dependencies
+#### Install the Driver (TT-KMD)
```
You copy pasted these from the other pages?

I think there needs to be more synchronization between our docs' team and the customer docs' team. There seems to be separate tracks of work trying to do the same thing. Let's set up some time between @zzigler-tt  and @bbeggsTT to work out a game plan and be clear with what our goals are.

I know the deliverable we want to provide a seamless experience for customers who are less hacker-inclined. I'll caution everyone in saying a couple points:

1) Multiple people + teams in TT want the same thing
2) For this particular deliverable, trying to most fast but separately from the organization will only result in a documentation that will leave customers still confused.
3) As a side note, copy-pasting documentation like this is something that SysEng (@warthog9 ) has specifically requested we *stop* and have an organization plan for.

bbeggsTT replied:
We included that part here because it is relatively short and considering @adifrancescoTT  points about creating self contained installation guides and cutting down on amount of clicks/link hopping needed through an installation procedure.

From @adifrancescoTT's presentation here: https://docs.google.com/presentation/d/1xT3UonEg3u_UHP2ehaBAZwRhKt30b2RlenmHyG10PRk/edit#slide=id.g31b7042d5d0_0_73

We did consult @warthog9 about including the TT-KMD section, I can add you to that conversation in slack.

Our options when including relevant information from other locations are:
- Link the Content - Creates disjointed feeling in documentation by going between too many locations for a single install procedure. Cuts down on document length.
- Include the Content - Cuts down on the amount of link/location hopping but will force us to keep track of updates in other locations as well our own. Can also make documents much longer and daunting.
- Include Small Sections with Link - This is the approach we took for this section. The relevant info is included but with a reference to the full content.
- Alternative Option - Can we create a custom action in GitHub that allows us to "snapshot" or "include" content from other locations? Something that auto-updates if the original content changes?

Hopefully this provides some clarity on our intentions for these changes. All feedback is welcome.

tt-rkim replied:
Sounds good, if you consulted with both @adifrancescoTT and @warthog9 's teams about this, then I'm good with approach 3.

blozano-tt commented on INSTALLING.md:
````diff
@@ -2,199 +2,266 @@
+- Download and install HugePages:
 ```sh
 wget https://raw.githubusercontent.com/tenstorrent/tt-metal/main/infra/machine_setup/scripts/setup_hugepages.py
+sudo -E python3 setup_hugepages.py first_pass
````
Note: maybe we should break out the script changes to a different PR. But let me know what you think of the optimization.

warthog9 replied:
Shouldn't the huge pages be migrating to the https://github.com/tenstorrent/tt-system-tools supported setup?

blozano-tt replied:
@warthog9 that was my intention.

The script downloads tt-system-tools release Debian and installs it.

Then starts the service. Anything incorrect here?

warthog9 replied:
hadn't looked, so that's on me.  That's solid till we get a proper package repo sorted (hopefully soon)

blozano-tt replied:
We can address my comments here in a follow up PR.
We are close to verifying and updating hugepages configuration to tt-system-tools flow.

gsarabandoTT committed:
move SW dependencies for codebase contributions

gsarabandoTT commented on CONTRIBUTING.md:
````diff
@@ -88,6 +88,15 @@ Then to check if Hugepages is enabled:
 python3 infra/machine_setup/scripts/setup_hugepages.py check
 ```
 
+### Software dependencies for codebase contributions
````
Moved this step from INSTALLING.md to CONTRIBUTING.md as it is specific to contributing.

tt-rkim replied:
Sure, works for me.

gsarabandoTT committed:
add release version note and link to tags

gsarabandoTT commented on INSTALLING.md:
```diff
@@ -2,199 +2,178 @@
 
 These instructions will guide you through the installation of Tenstorrent system tools and drivers, followed by the installation of TT-Metalium and TT-NN.
 
-> [!IMPORTANT]
```
Thanks, moved it back to the top and included link to version tags.

gsarabandoTT committed:
update compatibility matrix

blozano-tt commented on CONTRIBUTING.md:
````diff
@@ -88,6 +88,15 @@ Then to check if Hugepages is enabled:
+1. Install dependencies
+
+```sh
+sudo apt install pandoc libtbb-dev libcapstone-dev pkg-config
````
I can't imagine why these things would be needed to contribute.

Contributing only requires pre-commit as far as I'm aware.

blozano-tt replied:
![Uploading image.png…]()

blozano-tt replied:
<img width="442" alt="image" src="https://github.com/user-attachments/assets/f944fdce-498d-4418-9cf9-463853e0e3ee" />

gsarabandoTT replied:
@tt-rkim can you weight in on this? that step is listed as an optional step for contributors in current version of INSTALLING.md.

tt-rkim replied:
This is for profiling, which developers often use.

Should we make an additional section for source users who want to use the profiler?

gsarabandoTT replied:
moved it back to INSTALLING.md as additional section for source install, does it make sense now?

tt-rkim commented on INSTALLING.md:
````diff
@@ -4,197 +4,271 @@ These instructions will guide you through the installation of Tenstorrent system
+- Note: Some advanced build configurations like unity builds require `CMake 3.20`.
+  To upgrade Cmake run:
+    ```sh
+    pip3 install cmake --upgrade
````
Why are we installing `cmake` into the python environment?

I would slightly prefer if it's a system dependency.

gsarabandoTT replied:
This method seemed to work easier (in VMs at least). Is this the script we should refer to? 

```
# Use a newer version of CMake than what is available from Canonical for 20.04
RUN apt -y update \
    && apt install -y --no-install-recommends ca-certificates gpg wget \
    && wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - | tee /usr/share/keyrings/kitware-archive-keyring.gpg >/dev/null \
    && echo 'deb [signed-by=/usr/share/keyrings/kitware-archive-keyring.gpg] https://apt.kitware.com/ubuntu/ focal main' | tee /etc/apt/sources.list.d/kitware.list >/dev/null \
    && rm -rf /var/lib/apt/lists/*
```

tt-rkim replied:
"Work easier"?

This might conflict with `install_dependencies.sh`, and also note that not all users will be in a Python environment, like certain customers we have in mind.

Please use `apt`

gsarabandoTT replied:
What I mean by "work easier" is that we were able to successfully install it with fewer commands and running into less issues during tests. I have no technical preference in this matter and will include the method your prefer. Just please let me know if the commands I included above are satisfactory.

tt-rkim replied:
I prefer we do not use `pip` install.

Try a

```
sudo apt update
sudo apt install cmake
```

to get newest cmake

ttmchiou replied:
FWIW we install cmake with apt install so it would follow what our current processes are as well

gsarabandoTT replied:
Thanks guys, update it in latest commit.

tt-rkim commented on INSTALLING.md:
```diff
@@ -4,197 +4,271 @@ These instructions will guide you through the installation of Tenstorrent system
-
-> [!NOTE]
->
-> Note about Python environments: You do not have to use `create_venv.sh`. If you
```
Why is this gone?

gsarabandoTT replied:
The reasoning was that if you are less familiar with Python and its various environment tools, you will run the recommended command, otherwise you will know you can build it in a custom environment and that you may run into compatibility issues (a user running into errors for doing something other than what is recommended seems out of our scope). 

The previous text had a few (non-technical) issues but if the reasoning is wrong let me know what information exactly you would like to see here and I’ll add to it.

tt-rkim replied:
Sure, I'm fine with that stance.

Users creating environments with methods outside of our recommendations is out of scope.

gsarabandoTT committed:
update cmake as system dependency

tt-rkim commented on INSTALLING.md:
```diff
@@ -4,197 +4,269 @@ These instructions will guide you through the installation of Tenstorrent system
 
-### Step 3. Hugepages
+#### Install TT-Flash
+- Install Cargo (Rust package manager):
```
I would recommend linting this.

tt-rkim commented on INSTALLING.md:
```diff
@@ -4,197 +4,269 @@ These instructions will guide you through the installation of Tenstorrent system
 
-### Step 2. System-level dependencies
+#### Install the Driver (TT-KMD)
+- DKMs must be installed:
```
Not all capitals?

gsarabandoTT replied:
Fixed, thank you for spotting it

bbeggsTT committed:
Update INSTALLING.md

Omit hugepages config step. Hugepages config is now included in dependency installation.

gsarabandoTT committed:
add profiling section for source users

gsarabandoTT committed:
update env variables use

gsarabandoTT commented on INSTALLING.md:
````diff
@@ -4,197 +4,255 @@ These instructions will guide you through the installation of Tenstorrent system
 
-### Step 5. (Optional) Software dependencies for codebase contributions
+- To verify your installation, try executing an example:
 
-Please follow the next additional steps if you want to contribute to the codebase.
+  ```
+  python3 -m ttnn.examples.usage.run_op_on_device
+  ```
 
-1. Install dependencies
+- Visit Tenstorrent's [TT-NN Basic Examples page](https://docs.tenstorrent.com/ttnn/latest/ttnn/usage.html#basic-examples) or get started with [simple kernels on TT-Metalium](https://docs.tenstorrent.com/tt-metalium/latest/tt_metal/examples/index.html).
 
````
@blozano-tt Could you please confirm these changes make sense?

blozano-tt replied:
Looks good for now. 

It pains me that we still have to have ARCH_NAME set for runtime. 

I think we should confirm that… and then aim to remove the requirement ASAP.

gsarabandoTT committed:
fix typo

tt-rkim commented:
Are we ok hre?

gsarabandoTT committed:
Merge branch 'main' into bbeggsTT-patch-5

gsarabandoTT committed:
update links

tt-rkim commented on INSTALLING.md:
````diff
@@ -4,123 +4,176 @@ These instructions will guide you through the installation of Tenstorrent system
+```
+git clone https://github.com/tenstorrent/tt-kmd.git
+cd tt-kmd
+git checkout -b ttkmd-1.29 ttkmd-1.29
````
This number is going to change as time goes on - be wary of that. So it may be out of sync with the table. Users may be confused in that case.

warthog9 replied:
worth looking at sphinx's templating and possibly parameterizing that so it can get updated easier?

warthog9 replied:
Flipping this on it's head some, is there a reason not to just point to the central quick start guide as "go do this and come back here", and/or just pull that in on your build?

tt-rkim replied:
> is there a reason not to just point to the central quick start guide 

It was originally this, but @bbeggsTT and the docs org wanted to reduce the pulls from build: https://github.com/tenstorrent/tt-metal/pull/15800#discussion_r1878650699

tt-rkim commented on INSTALLING.md:
```diff
@@ -4,123 +4,176 @@ These instructions will guide you through the installation of Tenstorrent system
+---
+
+### Option 2: From Docker Release Image
+Installing from Docker Release Image is the quickest way to access our APIs and to start runnig AI models.
```
```suggestion
Installing from Docker Release Image is the quickest way to access our APIs and to start running AI models.
```

tt-rkim commented on INSTALLING.md:
```diff
@@ -134,26 +187,71 @@ When inside of the container,
+---
+
+### Option 3: From Wheel
+Instal from wheel for quick access to our APIs and to get an AI model running
```
```suggestion
Install from wheel for quick access to our APIs and to get an AI model running
```

tt-rkim reviewed (approved):
Approved, but note the following:

Things for you to-do:
- Respond to any comments/suggestions I made
- Maybe proof-read or feed into a spellchecker once to make sure we don't miss anything

Things for you to acknowledge:
- Someone will need to be responsible for updating the SysEng-related portions of the full install and that it's in sync with what the SysEng org has written. I'm assuming your team will be doing this?

gsarabandoTT committed:
Update INSTALLING.md

Co-authored-by: Raymond Kim <109366641+tt-rkim@users.noreply.github.com>

gsarabandoTT committed:
Update INSTALLING.md

Co-authored-by: Raymond Kim <109366641+tt-rkim@users.noreply.github.com>

gsarabandoTT committed:
check spelling

gsarabandoTT committed:
update INSTALLING.md

gsarabandoTT committed:
update INSTALLING.md
